src/backend/metal/src/device.rs

```diff
@@ -857,6 +884,26 @@ impl hal::Device<Backend> for Device {
                             mtl_textures.extend(images.iter().map(|image| &*((image.0).0)));
                             encoder.set_textures(&mtl_textures, write.binding as _);
                         },
+                        UniformBuffer(ref buffers) | StorageBuffer(ref buffers) => {
+                            mtl_buffers.clear();
+                            mtl_buffers.extend(buffers.iter().map(|buffer| &*((buffer.0).0)));
+                            mtl_offsets.clear();
+                            mtl_offsets.extend(buffers.iter().map(|buffer| buffer.1.clone()));
+
+                            let encoder: &metal::ArgumentEncoderRef = &encoder;
+
+                            let range = NSRange {
+                                location: offset,
+                                length: mtl_buffers.len() as NSUInteger,
+                            };
+                            unsafe {
+                                msg_send![encoder,
```

Comment:
can we call it as `encoder.set_buffers` instead of sending a custom message?

Reply:
Should I open PR in metal repo or just add function here?

Reply:
eh, you are correct. Will leave this for follow up work.